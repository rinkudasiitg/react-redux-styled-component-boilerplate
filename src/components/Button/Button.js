import React from 'react';
import styled, { css } from 'styled-components';

const Button = styled.button`
    background: ${props => props.theme.colorPrimary};
    border: 5px solid ${props => props.theme.colorGrayDark};
    border-radius: 50px;
    font-size: 20px;
    color: ${props => props.theme.colorSecondary};
    outline: none;
    
    ${props => props.primary && css`
      background: yellow;
    `}
`;

export default Button;