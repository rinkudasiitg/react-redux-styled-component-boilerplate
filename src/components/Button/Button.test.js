import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

test('should render the Button correctly', () => {
  const wrapper = shallow(<Button />);
  expect(wrapper).toMatchSnapshot();
});

test('should render the Button with the attribute primary', () => {
    const wrapper = shallow(<Button primary/>);
    expect(wrapper).toMatchSnapshot();
});
