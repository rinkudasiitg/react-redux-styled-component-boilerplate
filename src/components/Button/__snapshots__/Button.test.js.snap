// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Button correctly 1`] = `
<button
  className="sc-bdVaJa cHdtTr"
/>
`;

exports[`should render the Button with the attribute primary 1`] = `
<button
  className="sc-bdVaJa kzCDOL"
/>
`;
